x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/rules/action/TransportTestQueryRulesetAction.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.rules.action;
+
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.support.ActionFilters;
+import org.elasticsearch.action.support.HandledTransportAction;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.client.internal.OriginSettingClient;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.util.concurrent.EsExecutors;
+import org.elasticsearch.injection.guice.Inject;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.transport.TransportService;
+import org.elasticsearch.xpack.application.rules.QueryRule;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.elasticsearch.xpack.core.ClientHelper.ENT_SEARCH_ORIGIN;
+
+public class TransportTestQueryRulesetAction extends HandledTransportAction<
+    TestQueryRulesetAction.Request,
+    TestQueryRulesetAction.Response> {
+
+    private final Client originSettingClient;
+
+    @Inject
+    public TransportTestQueryRulesetAction(
+        TransportService transportService,
+        ClusterService clusterService,
+        ActionFilters actionFilters,
+        Client client
+    ) {
+        super(
+            TestQueryRulesetAction.NAME,
+            transportService,
+            actionFilters,
+            TestQueryRulesetAction.Request::new,
+            EsExecutors.DIRECT_EXECUTOR_SERVICE
+        );
+        this.originSettingClient = new OriginSettingClient(client, ENT_SEARCH_ORIGIN);
+    }
+
+    @Override
+    protected void doExecute(Task task, TestQueryRulesetAction.Request request, ActionListener<TestQueryRulesetAction.Response> listener) {
+        GetQueryRulesetAction.Request getQueryRulesetRequest = new GetQueryRulesetAction.Request(request.rulesetId());
+        originSettingClient.execute(GetQueryRulesetAction.INSTANCE, getQueryRulesetRequest, new ActionListener<>() {
```

Comment:
If we ever want to open this API to not require the `manage_search_query_rules` privilege like we discussed, if changing this similarly to https://github.com/elastic/elasticsearch/pull/105667 to use `executeAsyncWithOrigin` would be the solution

Reply:
Thanks, I will look into that!

Reply:
Believe it or not, that doesn't work either! 

I updated the call to use `executeAsyncWithOrigin` and I still get yaml tests failing with the following error: 
```
- type: "security_exception"
        reason: "action [indices:data/read/xpack/query_rules/test] is unauthorized for\
          \ user [entsearch-admin] with effective roles [admin] on restricted indices\
          \ [.query-rules], this action is granted by the index privileges [read,all]"
```

I am stumped why this doesn't work, as it definitely does when we use the rule query. I'm inclined to punt and figure it out later before we GA the call, just to get something out there. That would require us keeping the manage query rules privilege for now. What are your thoughts?